libs/ledger-live-common/src/families/evm/editTransaction/getEditTransactionPatch.ts

```diff
@@ -0,0 +1,264 @@
+     */
+    gasOptions,
+  };
+
+  return patch;
+};
+
+/**
+ * Can't easily and properly use generics with Partial to avoid type casting
+ * when used with EvmTransactionLegacy or EvmTransactionEIP1559
+ * cf. https://github.com/microsoft/TypeScript/issues/13442
+ * Ideal signature would be something like:
+ * getCommonCancelPatch<T extends Transaction>(...): Partial<T>
+ * With T being either EvmTransactionLegacy or EvmTransactionEIP1559
+ */
+const getCommonCancelPatch = ({
+  transaction,
+  shouldUseFastStrategy,
+  freshAddress,
+  gasOptions,
+}: {
+  transaction: Transaction;
+  shouldUseFastStrategy: boolean;
+  freshAddress: string;
+  gasOptions: GasOptions;
+}): Partial<Transaction> => {
+  const patch: Partial<Transaction> = {
+    amount: new BigNumber(0),
+    // removing the subAccountId to transform a token tx to main coin tx
+    subAccountId: undefined,
+    data: undefined,
+    nonce: transaction.nonce,
+    mode: "send",
+    recipient: freshAddress,
+    feesStrategy: shouldUseFastStrategy ? "fast" : "custom",
+    useAllAmount: false,
+    gasLimit: transaction.gasLimit,
+    customGasLimit: transaction.gasLimit,
+    /**
+     * we add gasOptions to the patch to prevent `prepareTransaction` to request
+     * value from a node when fast fee option is used
+     * cf. https://github.com/LedgerHQ/ledger-live/blob/f03479a40eba734e1ac4ec92415fe139b3b7eee6/libs/coin-evm/src/prepareTransaction.ts#L200-L201
+     */
+    gasOptions,
+  };
+
+  return patch;
+};
+
+const getLegacyPatch = ({
+  transaction,
+  gasOptions,
+  editType,
+  account,
+}: {
+  transaction: EvmTransactionLegacy;
+  gasOptions: GasOptions;
+  editType: EditType;
+  account: Account;
+}): Partial<EvmTransactionLegacy> => {
+  const fastFeeData = gasOptions.fast;
+
+  if (!fastFeeData.gasPrice) {
+    throw new Error("gasOptions.fast.gasPrice is required");
+  }
+
+  // we get the minimum possible fees values needed to update the transaction
+  const { gasPrice: minGasPrice } = getMinLegacyFees({ gasPrice: transaction.gasPrice });
+
+  /**
+   * Depending on the current network conditions, the minimum fees values might
+   * not be enought to properly update the transaction. The updated fees might
+   * still be lower than the current network fees.
+   * To prevent this, which would lead to an updated transaction still being stuck,
+   * we use the maximum between the minimum fees values and the current network fees,
+   * as a default value.
+   */
+  const gasPrice = BigNumber.max(minGasPrice, fastFeeData.gasPrice);
+
+  const shouldUseFastStrategy = gasPrice.isEqualTo(fastFeeData.gasPrice);
+
+  if (editType === "speedup") {
+    const patch: Partial<EvmTransactionLegacy> = {
+      ...(getCommonSpeedUpPatch({
+        transaction,
+        shouldUseFastStrategy,
+        gasOptions,
+      }) as Partial<EvmTransactionLegacy>),
+      maxFeePerGas: undefined,
+      maxPriorityFeePerGas: undefined,
```

Comment:
maybe the 

maxFeePerGas: undefined,
maxPriorityFeePerGas: undefined,

are not necessary? as you only need to return a Partial transaction.
same for other similar pattern in this file